{Release} Upgrade to Azure CLI 2.46.0 (#25663)
Azure/azure-cli: src/azure-cli/HISTORY.rst

Release History
===============

2.46.0
++++++

**ACR**

> * `az acr token`, `az acr scope-map`: Tokens and Scope-Maps are Generally Available (#25294)

Reviewer: ```suggestion
* `az acr token\scope-map`: Tokens and Scope-Maps are Generally Available (#25294)
```

* `az acr manifest`: Support oci image index (#25353)
* `az acr cache`: Add new command group to manage cache rules (#25096)
* `az acr credential-set`: Add new command group to manage credential sets (#25096)
* Fix #24886: `az acr`: Improve the 429 error handling for CONNECTIVITY_REFRESH_TOKEN_ERROR (#25549)

**AKS**

* `az aks check-acr`: Fix mariner node missing cert (#25453)

> * Fix #25521: `az aks nodepool upgrade` crashes when the version passed in through the parameter `--kubernetes-version` is the same as the cluster version (#25526)

Reviewer: ```suggestion
* Fix #25521: `az aks nodepool upgrade`: Fix the crashes when the version passed in through the parameter `--kubernetes-version` is the same as the cluster version (#25526)
```

* [BREAKING CHANGE] `az network cross-region-lb`: Remove in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles. (#25494)
* [BREAKING CHANGE] `az network nic ip-config`: Rename output property `privateIpAddress` to `privateIPAddress` to keep consistent with the name in API (#25498)
* [BREAKING CHANGE] `az network nic ip-config`: Rename output property `privateIpAllocationMethod` to `privateIPAllocationMethod` to keep consistent with the name in API (#25498)
* [BREAKING CHANGE] `az network local-gateway`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25513)
* [BREAKING CHANGE] `az network vpn-connection`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25536)
* [BREAKING CHANGE] `az network vnet-gateway`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25588)

> * [BREAKING CHANGE] `az network watcher flow-log`: Remove the deprecated command `configure` (#25559)

Reviewer 2: ```suggestion
* [BREAKING CHANGE] `az network nic`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile (#25645)
* [BREAKING CHANGE] `az network watcher flow-log`: Remove the deprecated command `configure` (#25559)
```

* [BREAKING CHANGE] `az network vrouter`: Deprecate `vrouter` and use `routeserver` instead (#25603)
* [BREAKING CHANGE] `az network watcher connection-monitor endpoint add`: Remove deprecated parameters `filter-item` and `filter-type` (#25518)
* `az network nsg rule list`: Fix `-o table` cannot be used (#25387)
* `az network private-endpoint-connection`: Add provider `Microsoft.Monitor/Accounts` (#25489)
* `az network express-route gateway connection create/update`: Add parameters `--inbound-route-map` and `--outbound-route-map` to support route map (#25493)
* Fix #25408: `az network application-gateway rule create`: Creation fails with `--redirect-config` when there are multiple pools (#25562)
* `az network private-endpoint-connection`: Add provider `Microsoft.DBforMySQL/flexibleServers` (#25422)

> 

Reviewer 2: ```suggestion
[BREAKING CHANGE] `az network nic`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile (#25645)

```

* `az functionapp create`: Add new parameter `--environment` to support setting the name of container app environment (#25223)
* `az webapp config ssl bind/unbind`: Allow user to specify hostname to (un)bind with `--hostname` (#25362)
* `az webapp config ssl create/upload/import`: Allow user to specify certificate name with `--certificate-name` (#25367)
* `az functionapp create`: Add new parameter `--environment` to support setting the name of container app environment (#25659)

**ARM**

* `az bicep publish`: Update command to support new optional parameter `--documentationUri` (#25461)
* Fix #25510: `az bicep`: Set bicep.use_binary_from_path to false when installed using Azure CLI (#25541)
* `az bicep format`: Add new command to support formatting a Bicep file (#24605)

**ARO**

* `az aro create`: Rename the create `install-version` parameter to `version` (#25511)

**Compute**

* `az vmss reimage`: Fix the bug that all instances will be reimaged after using `--instance-id` and add new parameter `--instance-ids` to replace `--instance-id` (#25477)
* `az vm create`: Support recommending more suitable regions through warning log when creating VM (#25529)

> * `az vm/vmss identity assign`: Add warning log and modify help to inform that the default value Contributor of `--role` will be removed (#25283)

Reviewer 3: ```suggestion
* `az vm/vmss identity assign`: Add warning and modify help message for `--role`: Please note that the default value of `--role` will be removed in the breaking change release of the fall of 2023, so please specify "--role" and "--scope" at the same time when assigning a role to the managed identity. (#25283)
```

* [BREAKING CHANGE] `az network lb address-pool`: Output properties `privateIpAddress`, `privateIpAddressVersion`, `privateIpAllocationMethod`, `publicIpAddress` and `publicIpPrefix` are renamed by `privateIPAddress`, `privateIPAddressVersion`, `privateIPAllocationMethod`, `publicIPAddress` and `publicIPPrefix` to keep consistent with the name in API (#25360)
* [BREAKING CHANGE] `az network cross-region-lb probe`: Deprecate command group as probes are not supported for global load balancer (#25386)
* [BREAKING CHANGE] `az network nic create/update`: Rename output property `enableIpForwarding` to `enableIPForwarding` to keep consistent with the name in API (#25454)
* [BREAKING CHANGE] `az network nic create/update`: Rename output property `privateIpAllocationMethod` to `privateIPAllocationMethod` to keep consistent with the name in API (#25454)
* [BREAKING CHANGE] `az network nic create/update`: Rename output property `publicIpAddress` to `publicIPAddress` to keep consistent with the name in API (#25454)

> * [BREAKING CHANGE] `az network lb`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25494)

Reviewer 4: ```suggestion
* [BREAKING CHANGE] `az network lb`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25494)
```

* [BREAKING CHANGE] `az network cross-region-lb`: Remove in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles. (#25494)
* [BREAKING CHANGE] `az network nic ip-config`: Rename output property `privateIpAddress` to `privateIPAddress` to keep consistent with the name in API (#25498)
* [BREAKING CHANGE] `az network nic ip-config`: Rename output property `privateIpAllocationMethod` to `privateIPAllocationMethod` to keep consistent with the name in API (#25498)

> * [BREAKING CHANGE] `az network local-gateway`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25513)
> * [BREAKING CHANGE] `az network vpn-connection`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25536)
> * [BREAKING CHANGE] `az network vnet-gateway`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25588)
> * [BREAKING CHANGE] `az network watcher flow-log`: Removed the deprecated command `configure` (#25559)

Reviewer 4: ```suggestion
* [BREAKING CHANGE] `az network local-gateway`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25513)
* [BREAKING CHANGE] `az network vpn-connection`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25536)
* [BREAKING CHANGE] `az network vnet-gateway`: Update output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25588)
* [BREAKING CHANGE] `az network watcher flow-log`: Remove the deprecated command `configure` (#25559)
```

**PolicyInsights**

* Fix #25538: `az policy remediation create`: Fix the `Required property 'policyAssignmentId' not found in JSON` error (#25575)

**RDBMS**

* `az postgres flexible-server migration`: Use Cloud supplied URL's rather than hardcoded management URL's (#25332)
* `az mysql flexible-server replica create`: Add `--location` to support specifying replica location (#25492)
* `az mysql flexible-server update`: Fix `--storage-auto-grow` parameter unable to be set (#25628)

**Role**

* `az role assignment create`: Show warning if `--scope` argument is not specified: `--scope` argument will become required for creating a role assignment in the breaking change release of the fall of 2023. Please explicitly specify `--scope`. (#24755)
* Migrate `azure-mgmt-authorization` SDK to Track 2 and bump API version to 2022-04-01 (#25452)

**Service Bus**

* `az servicebus namespace`: Add `--premium-messaging-partitions` to support ServiceBus Namespace ScaleSet (#24835)

**Service Connector**

* `az connection create`: Add new param `--customized-keys` (#25515)

**SQL**

* `az sql instance-failover-group`: Add `--secondary-type` parameter to create and update commands (#25531)

> * `az sql midb restore`: Add tags parameter (#25519)

Reviewer 5: ```suggestion
* `az sql midb restore`: Add `--tags` parameter (#25519)
```

* `az functionapp create`: Add new parameter `--environment` to support setting the name of container app environment (#25223)
* `az webapp config ssl bind/unbind`: Allow user to specify hostname to (un)bind with `--hostname` (#25362)
* `az webapp config ssl create/upload/import`: Allow user to specify certificate name with `--certificate-name` (#25367)
* `az functionapp create`: Add new parameter `--environment` to support setting the name of container app environment (#25659)

**ARM**

* `az bicep publish`: Update command to support new optional parameter `--documentationUri` (#25461)
* Fix #25510: `az bicep`: Set bicep.use_binary_from_path to false when installed using Azure CLI (#25541)
* `az bicep format`: Add new command to support formatting a Bicep file (#24605)

**ARO**

* `az aro create`: Rename the create `install-version` parameter to `version` (#25511)

**Compute**

* `az vmss reimage`: Fix the bug that all instances will be reimaged after using `--instance-id` and add new parameter `--instance-ids` to replace `--instance-id` (#25477)
* `az vm create`: Support recommending more suitable regions through warning log when creating VM (#25529)

> * `az vm/vmss identity assign`: Add warning log and modify help to inform that the default value Contributor of `--role` will be removed (#25283)

Reviewer: ```suggestion
* `az vm/vmss identity assign`: Add warning log and modify help message to inform that the default value Contributor of `--role` will be removed, please specify `--role` and `--scope` at the same time when assigning a role to the managed identity (#25283)
```

* [BREAKING CHANGE] `az network cross-region-lb probe`: Deprecate command group as probes are not supported for global load balancer (#25386)
* [BREAKING CHANGE] `az network nic create/update`: Rename output property `enableIpForwarding` to `enableIPForwarding` to keep consistent with the name in API (#25454)
* [BREAKING CHANGE] `az network nic create/update`: Rename output property `privateIpAllocationMethod` to `privateIPAllocationMethod` to keep consistent with the name in API (#25454)
* [BREAKING CHANGE] `az network nic create/update`: Rename output property `publicIpAddress` to `publicIPAddress` to keep consistent with the name in API (#25454)
* [BREAKING CHANGE] `az network lb`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25494)

> * [BREAKING CHANGE] `az network cross-region-lb`: Removed in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles. (#25494)

Reviewer 4: ```suggestion
* [BREAKING CHANGE] `az network cross-region-lb`: Remove in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles. (#25494)
```

* [BREAKING CHANGE] `az network local-gateway`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25513)
* [BREAKING CHANGE] `az network vpn-connection`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25536)
* [BREAKING CHANGE] `az network vnet-gateway`: Updated output property names in `2017-03-09-profile`, `2018-03-01-hybrid`, `2019-03-01-hybrid` and `2019-03-01-hybrid` profiles to keep consist with the `latest` profile. (#25588)
* [BREAKING CHANGE] `az network watcher flow-log`: Removed the deprecated command `configure` (#25559)
* [BREAKING CHANGE] `az network vrouter`: Deprecate `vrouter` and use `routeserver` instead (#25603)
* [BREAKING CHANGE] `az network watcher connection-monitor endpoint add`: Remove deprecated parameters `filter-item` and `filter-type` (#25518)
* `az network nsg rule list`: Fix `-o table` cannot be used (#25387)

> * `az network watcher`: Add network watcher base cmd (#25436)
> * `az network watcher connection-monitor`: Add network watcher connection-monitor cmd (#25436)

Reviewer 4: ```suggestion
```

> * `az network private-endpoint-connection`: Add Provider Microsoft.Monitor/Accounts (#25489)

Reviewer 4: ```suggestion
* `az network private-endpoint-connection`: Add provider `Microsoft.Monitor/Accounts` (#25489)
```
